x-pack/plugins/monitoring/public/plugin.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import './views/all';
+import { i18n } from '@kbn/i18n';
+import {
+  App,
+  AppMountParameters,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from 'kibana/public';
+import {
+  FeatureCatalogueCategory,
+  HomePublicPluginSetup,
+} from '../../../../src/plugins/home/public';
+import { initAngularBootstrap } from '../../../../src/plugins/kibana_legacy/public';
+import { DEFAULT_APP_CATEGORIES } from '../../../../src/core/utils';
+import { MonitoringPluginDependencies, MonitoringConfig } from './types';
+import {
+  MONITORING_CONFIG_ALERTING_EMAIL_ADDRESS,
+  KIBANA_ALERTING_ENABLED,
+} from '../common/constants';
+
+export class MonitoringPlugin
+  implements Plugin<void, void, MonitoringPluginDependencies, MonitoringPluginDependencies> {
+  constructor(private initializerContext: PluginInitializerContext<MonitoringConfig>) {}
+
+  public setup(
+    core: CoreSetup<MonitoringPluginDependencies>,
+    plugins: object & { home?: HomePublicPluginSetup; cloud?: { isCloudEnabled: boolean } }
+  ) {
+    const { home } = plugins;
+    const id = 'monitoring';
+    const icon = 'monitoringApp';
+    const path = '/app/monitoring';
+    const title = i18n.translate('xpack.monitoring.stackMonitoringTitle', {
+      defaultMessage: 'Stack Monitoring',
+    });
+    const monitoring = this.initializerContext.config.get();
```

Comment:
Can you go ahead and return the `ui.enabled` config from this setup function so that I can integrate with it from the logstash plugin?

Reply:
Fixed, but this got me thinking about the following https://github.com/elastic/kibana/issues/64599

Reply:
Actually looking at how this value is used in the Logstash plugin, I wonder if we even need this. It currently uses the `monitoring.ui.enabled` flag to gate whether or not we hit the monitoring _backend_ for a list of logstash pipelines being tracked by logstash's monitoring data.

This seems rather roundabout to me, I would expect that we could / should get this list of pipelines from monitoring if the monitoring plugin is available at all, rather than only if the ui is enabled. @ycombinator does this make sense to you?

Reply:
@joshdover what you are saying makes sense to me. The purpose of the check in the `logstash` plugin was to make sure it didn't call a monitoring HTTP API route that wasn't even going to be registered. 

I don't recall why it was relying on the `monitoring.ui.enabled` flag for this. The only reasons I can think of is that that flag was also being used to gate whether the monitoring API routes were being registered or not AND that it was the only flag available on the client-side for the `logstash` plugin's client-side code to check. 

Ultimately, the _intent_ is that the `logstash` plugin should be able to get the list of pipelines from the `monitoring` plugin via an API (HTTP or JS). However, if said API is absent, the `logstash` plugin shouldn't even try to call it.

Reply:
Okay, in this case I think the safest option is to do something like: 
```javascript
if (!monitoring.ui.enabled || !monitoring.enabled) {
    return false;
}
...
return true;
```

Doesn't look like we're doing anything with `monitoring.enabled` on the backend, but maybe ES is?

Reply:
Ok I think the best solution here is to have the monitoring plugin simply expose a JS API on the frontend, something like `getLogstashPipelines` that the logstash plugin can call if/when the monitoring plugin is available. Let's do this as a follow up to this PR.